sklearn/linear_model/least_angle.py

```diff
@@ -150,28 +387,16 @@ def lars_path(X, y, Xy=None, Gram=None, max_iter=500,
                       ' This option will be removed in version 0.22.',
                       DeprecationWarning)
 
-    n_features = X.shape[1]
-    n_samples = y.size
-    max_features = min(max_iter, n_features)
+    if not n_samples:
+        if y is not None:
+            n_samples = y.size
+        else:
+            raise ValueError('y cannot be None while n_samples is None.')
```

Comment:
you should now support only the case n_samples is available. Unless it breaks backward compat.